azurelinuxagent/common/future.py

```diff
@@ -109,6 +109,12 @@ def get_linux_distribution_from_distro(get_full_name):
     )
     full_name = distro.linux_distribution()[0].strip()
     osinfo.append(full_name)
+
+    # Fixing is the problem https://github.com/Azure/WALinuxAgent/issues/2715. Distro.linux_distribution method not retuning full version
+    # If best is true, the most precise version number out of all examined sources is returned.
+    if osinfo and "mariner" in osinfo[0].lower():
```

Comment:
is this a check for None on distro? is it needed?

Reply:
Looks like in this context, it's not needed. Initially thought, we may have empty list but above we are appending some value. So, it will never be empty. I'll update